### PR TITLE
Add general test to avoid in-place modification of metadata by interfaces 

### PR DESCRIPTION
**Pull request by h-mayorquin**
OK, maybe something like this?

### CODE REVIEW

File: src/neuroconv/tools/testing/data_interface_mixins.py [hunk 1]
```diff
@@ -123,6 +123,27 @@ def test_conversion_as_lone_interface(self):
                 # Any extra custom checks to run
                 self.run_custom_checks()
 
+    def test_add_to_nwb(self):
```
Comment (CodyCBakerPhD) [inline src/neuroconv/tools/testing/data_interface_mixins.py:126]:
```suggestion
    def test_metadata_immutability(self):
```

Comment (CodyCBakerPhD) [reply src/neuroconv/tools/testing/data_interface_mixins.py:126]: or 'test_no_metadata_mutation'

**Comment by CodyCBakerPhD** [comment]
Yeah this is exactly what I was thinking

File: src/neuroconv/tools/testing/data_interface_mixins.py (hunk 1 above)
Comment (h-mayorquin) [reply src/neuroconv/tools/testing/data_interface_mixins.py]: I like both.

Comment (h-mayorquin) [reply src/neuroconv/tools/testing/data_interface_mixins.py]: Did something like the second suggestion and added interface.

**Comment by h-mayorquin** [comment]
There is some redundancy in this test as we don't need to check this for every case I feel. We could use our mock interfaces for some cases to avoid that excess but this should suffice for now.

**Comment by CodyCBakerPhD** [comment]
> There is some redundancy in this test as we don't need to check this for every case I feel. 

An interface could override `get_metadata` (as they often do) and some unintentional aspect of that could do the same problem anywhere at any time, so I'd be in favor of exhaustive testing

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Oh my, looks like more fixes are needed

**Comment by h-mayorquin** [comment]
OK, this is is a bigger problem that I don't have time to solve. Some of the formats (EDF) don't allow for concurrent access and either within the same test or because of pytest concurrency this is happening. Any suggestions so we can delay having to solve this at the format level?

**Comment by CodyCBakerPhD** [comment]
> OK, this is is a bigger problem that I don't have time to solve. Some of the formats (EDF) don't allow for concurrent access and either within the same test or because of pytest concurrency this is happening. Any suggestions so we can delay having to solve this at the format level?

Interesting!

Maybe make it a `check_` instead of a `test_` and inject it into the main mixing test call?

**Comment by h-mayorquin** [comment]
> > OK, this is is a bigger problem that I don't have time to solve. Some of the formats (EDF) don't allow for concurrent access and either within the same test or because of pytest concurrency this is happening. Any suggestions so we can delay having to solve this at the format level?
> 
> Interesting!
> 
> Maybe make it a `check_` instead of a `test_` and inject it into the main mixing test call?

Ended up doing something like this. I don't like it much but practice beats purity I guess.

**Pull request merged**